addons/medical_statemachine/functions/fnc_handleStateUnconscious.sqf

```diff
@@ -28,14 +27,22 @@ if (_painLevel > 0) then {
     [QEGVAR(medical,moan), [_unit, _painLevel]] call CBA_fnc_localEvent;
 };
 
-// Handle spontaneous wakeup from unconsciousness
+// Handle spontaneous wake up from unconsciousness
 if (EGVAR(medical,spontaneousWakeUpChance) > 0) then {
     if (_unit call EFUNC(medical_status,hasStableVitals)) then {
-        private _lastWakeUpCheck = _unit getVariable [QEGVAR(medical,lastWakeUpCheck), CBA_missionTime];
+        private _lastWakeUpCheck = _unit getVariable [QEGVAR(medical,lastWakeUpCheck);
+
+        // Handle setting being changed mid-mission and still properly check
+        // already unconscious units, should handle locality changes as well
+        if (isNil "_lastWakeUpCheck") exitWith {
+            _unit setVariable [QGVAR(medical,lastWakeUpCheck), CBA_missionTime];
```

Comment:
Could also trigger a wake up check right away rather than having the 15 second wait in this situation?

Reply:
I think it's good this way. That would introduce the possibility of going unconscious for less time than the check interval.

Reply:
this only applies to when the setting is changed mid-mission from 0% chance, 
so I think it's just fine